### PR TITLE
Vet clinic database: add 'join table' for visits

### PR DESCRIPTION
**Pull request by ernestkorir**
In this PR:

1. Create table `vets`
2. Create many to many relationships by adding a `specializations` table
3. Insert data to `Vets`, `specializations` and `visits`
4. Add Queries to find:

> Who was the last animal seen by William Tatcher?

244374245-41458031-14bc-41e1-9d55-e2ea623db0ed

> How many different animals did Stephanie Mendez see?
244374284-9a3895f1-3590-40a1-bcca-d689e7035a78

>  List all vets and their specialties, including vets with no specialties.

244374319-4cd2c094-10ac-46a3-a03d-4fc283f407f3

> List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.

244389156-c465d509-512e-4f4c-bcf4-1c6fd51e1060

> What animal has the most visits to vets
244374438-c719628c-c39a-4a8c-a39f-f1881b0938bf

> Who was Maisy Smith's first visit?

244374502-042574fb-0413-48bd-b7b1-96dc51f872ac

> Details for most recent visit: animal information, vet information, and date of visit.
244374550-07ea3386-b419-4265-ad93-e0f2c4a32bc7

> How many visits were with a vet that did not specialize in that animal's species?
244374581-6b50980a-83bd-47e5-970d-d5bae3dd77ee

>  What specialty should Maisy Smith consider getting? Look for the species she gets the most

244374676-a3808726-db14-462e-b1ef-cc79edc63a3f

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -123,4 +123,76 @@ FROM animals
...
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'William Tatcher' 
+ORDER BY v.visit_date DESC 
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT v.animal_id) 
+FROM visits v 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'Stephanie Mendez';
```
Comment (Kamba56) [inline queries.sql]: - [ ] Stephanie mendez actually visited 4 different animals, but from your screenshot, I can see your query returned 3. THis is due to incorrect data added to the visits table. Kindly fix data in the visits table.
image

File: data.sql [hunk 2]
```diff
@@ -46,4 +46,42 @@ SET owner_id =
...
+  ((SELECT id FROM vets WHERE name = 'Jack Harkness'), (SELECT id FROM species WHERE name = 'Digimon'));
+  
+-- Inserting data in the visits table
+INSERT INTO visits (animal_id, vet_id, visit_date) VALUES
+(1, 1, '2020-05-24'),
+(1, 3, '2020-07-22'),
+(2, 4, '2021-02-02'),
+(7, 2, '2020-01-05'),
+(7, 2, '2020-03-08'),
+(7, 2, '2020-05-14'),
```
Comment (Kamba56) [inline data.sql]: - [ ] This animal visited by Maisy Williams is supposed to be `Pikachu` with animal_id of `3`. The animal with Id 7 is Squirtle.

File: data.sql [hunk 3]
```diff
@@ -46,4 +46,42 @@ SET owner_id =
...
+(3, 3, '2021-05-04'),
```
Comment (Kamba56) [inline data.sql]: - [ ] The animal visited by Stephanie Mendez here is `Devimon` with animal_id `4`. The animal with id 3 is Pikachu. Kindly fix this please.

File: data.sql [hunk 4]
```diff
@@ -46,4 +46,42 @@ SET owner_id =
...
+(8, 4, '2021-02-24'),
```
Comment (Kamba56) [inline data.sql]: - [ ] The animal visited by Jack Harness here is `Chamander` with animal_id `5`. The animal with id 8 is Angemon. Kindly fix this please.

File: data.sql [hunk 5]
```diff
@@ -46,4 +46,42 @@ SET owner_id =
...
+(4, 1, '2020-08-10'),
+(4, 2, '2021-04-07'),
```
Comment (Kamba56) [inline data.sql]: - [ ] The animal visited here is `Plantmon` with animal_id `6`. The animal with id 4 is Devimon. Kindly fix this, please.
- [ ] ALso, you forgot to add one more visit to plantmon by Maisy Williams on `2019-12-21`.
image

File: data.sql [hunk 6]
```diff
@@ -46,4 +46,42 @@ SET owner_id =
...
+(9, 3, '2019-09-29'),
```
Comment (Kamba56) [inline data.sql]: - [ ] The animal visited here is `Squirtle` with animal_id `7`. The animal with id 9 is Boarmon. Kindly fix this, please.

File: data.sql [hunk 7]
```diff
@@ -46,4 +46,42 @@ SET owner_id =
...
+(5, 4, '2020-10-03'),
+(5, 4, '2020-11-04'),
```
Comment (Kamba56) [inline data.sql]: - [ ] The animal visited by Jack Harness here is `Angemon` with animal_id `8`. The animal with id 5 is Chamander. Kindly fix this, please.

File: data.sql [hunk 8]
```diff
@@ -46,4 +46,42 @@ SET owner_id =
...
+(6, 4, '2005-06-06'),
+(6, 2, '2019-01-24'),
+(6, 2, '2019-05-15'),
+(6, 2, '2020-02-27'),
```
Comment (Kamba56) [inline data.sql]: - [ ] The animal visited by Maisy Smiath here is `Boarmon` with animal_id `9`. The animal with id 6 is Plantmon. Kindly fix this, please.

File: data.sql [hunk 9]
```diff
@@ -46,4 +46,42 @@ SET owner_id =
...
+(6, 2, '2020-08-03'),
```
Comment (Kamba56) [inline data.sql]: - [ ] The animal visited by Stephanie Mandez here is `Blossom` with animal_id `10`. The animal with id 6 is Plantmon. Kindly fix this, please.

- [ ] Incorrect data added to the visits table is making the queries return incorrect results. Kindly fix the data added to the visits table and run queries again so you get correct results and add screenshots of the correct results.

File: queries.sql [hunk 10]
```diff
@@ -123,4 +123,76 @@ FROM animals
...
+LEFT JOIN species s ON s.id = sp.species_id
+ORDER BY vt.name;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name 
+FROM animals a 
+JOIN visits v ON a.id = v.animal_id 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'Stephanie Mendez' 
+AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
```
Comment (Kamba56) [inline queries.sql]: - [ ] This query is also returning incorrect result due to incorrect data in the visits table. Kindly fix data and add a new screenshot of correct result.

File: queries.sql [hunk 11]
```diff
@@ -123,4 +123,76 @@ FROM animals
...
+
+-- What animal has the most visits to vets?
+SELECT a.name AS animal_name, COUNT(*) AS visit_count 
+FROM animals a 
+JOIN visits v ON a.id = v.animal_id 
+GROUP BY a.id 
+ORDER BY visit_count DESC 
+LIMIT 1;
```
Comment (Kamba56) [inline queries.sql]: - [ ] This query is also returning incorrect results due to incorrect data in the visits table. Kindly fix the data and add a new screenshot of the correct result.

File: queries.sql [hunk 12]
```diff
@@ -123,4 +123,76 @@ FROM animals
...
+
+-- Who was Maisy Smith's first visit?
+SELECT a.name AS animal_name, MIN(v.visit_date) AS first_visit
+FROM animals a
+INNER JOIN visits v ON a.id = v.animal_id
+INNER JOIN vets vt ON vt.id = v.vet_id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY a.name;
```
Comment (Kamba56) [inline queries.sql]: - [ ] Maisy smiths first visit should be one animal and its Boarmon. Kindly fix the query and add a screenshot of the new result.

File: queries.sql [hunk 13]
```diff
@@ -123,4 +123,76 @@ FROM animals
...
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT a.*, v.*, MAX(visits.visit_date) AS most_recent_visit_date
+FROM visits
+JOIN animals a ON visits.animal_id = a.id
+JOIN vets v ON visits.vet_id = v.id
+WHERE visits.visit_date = (SELECT MAX(visit_date) FROM visits)
+GROUP BY a.id, v.id
```
Comment (Kamba56) [inline queries.sql]: - [ ] This query is also returning incorrect results due to incorrect data in the visits table. Kindly fix the data and add a new screenshot of the correct result.

File: queries.sql [hunk 14]
```diff
@@ -123,4 +123,76 @@ FROM animals
...
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT a.species_id, COUNT(*) AS num_visits
+FROM visits v
+INNER JOIN animals a ON v.animal_id = a.id
+WHERE v.vet_id = (
+  SELECT id FROM vets WHERE name = 'Maisy Smith'
+)
+GROUP BY a.species_id
+ORDER BY num_visits DESC
 LIMIT 1;
```
Comment (Kamba56) [inline queries.sql]: - [Optional] Consider returning the species name instead of the species id.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @ernestkorir, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job adding vets and specialization tables!👏👏👏
- [x] Great job using Join clause to query multiple tables.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
Hi @ernestkorir , 

### Highlights ✨ 

- Thanks for making the requested changes 👍 
- 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**